frameworks/Elixir/phoenix/web/controllers/page_controller.ex

```diff
@@ -50,32 +49,41 @@ defmodule Hello.PageController do
   end
 
   def updates(conn, params) do
-    q = try do
-      case String.to_integer(params["queries"]) do
-        x when x < 1    -> 1
-        x when x > 500  -> 500
-        x               -> x
-      end
-    rescue
-      ArgumentError -> 1
-    end
-
-    data = for _ <- 1..q do
-      id = :rand.uniform(10000)
-      num = :rand.uniform(10000)
-      w = Repo.get(World, id)
-      changeset = Ecto.Changeset.change(w, randomnumber: num)
-      Repo.update!(changeset)
-    end
+    json =
+      params["queries"]
+      |> query_range()
+      |> parallel(fn _ -> Repo.get(World, :rand.uniform(10000)) end)
+      |> parallel(fn w ->
```

Comment:
We should probably merge those two parallel into a single one You can also wrap both queries into a Repo.checkout in Ecto 3.0+.

Reply:
I was honestly considering an upsert, but I didn't know if that was in the spirit of the test. I guess since it is a feature of Ecto it _should_ be ok.  Will refactor it in a little bit.

Reply:
FYI, per the test requirements for updates, you must query the world from the database, update the random number on that object, and then update the database.

Reply:
Ok, won't change.